File: Lecture 3/FinalCode - Java/app/src/main/java/com/inovaeg/androidtrainingpharos2020/examples/CreateViewsAtRuntimeActivity.java
package com.inovaeg.androidtrainingpharos2020.examples;

import android.os.Bundle;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.inovaeg.androidtrainingpharos2020.R;

public class CreateViewsAtRuntimeActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_create_views_at_runtime);

        LinearLayout rootLinearLayout = findViewById(R.id.linear_create_view_at_runtime);

        TextView newTextView = new TextView(this);
        newTextView.setText("Hello world");
        newTextView.setBackgroundColor(getResources().getColor(R.color.colorAccent));

        LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(200, 200);
        layoutParams.setMargins(50, 50, 50, 50);
        rootLinearLayout.addView(newTextView, layoutParams);
    }
}

File: Lecture 3/FinalCode - Kotlin/app/src/main/java/com/inovaeg/androidtrainingpharos2020/examples/CreateViewsAtRuntimeActivity.kt
package com.inovaeg.androidtrainingpharos2020.examples

import android.os.Bundle
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.inovaeg.androidtrainingpharos2020.R
import kotlinx.android.synthetic.main.activity_create_views_at_runtime.*

class CreateViewsAtRuntimeActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_create_views_at_runtime)
        linear_create_view_at_runtime.addView(TextView(this).apply {
            text = "Hello world"
            setBackgroundColor(ContextCompat.getColor(this@CreateViewsAtRuntimeActivity, R.color.colorAccent))
        }, LinearLayout.LayoutParams(200, 200).apply {
            setMargins(50, 50, 50, 50)
        }
        )
    }
}